apiserver/facades/agent/storageprovisioner/storageprovisioner.go

```diff
@@ -1817,171 +1016,39 @@ func (s *StorageProvisionerAPIv4) SetFilesystemAttachmentInfo(
 	ctx context.Context,
 	args params.FilesystemAttachments,
 ) (params.ErrorResults, error) {
-	canAccess, err := s.getAttachmentAuthFunc(ctx)
-	if err != nil {
-		return params.ErrorResults{}, err
-	}
+	// TODO: implement this method using the storageProvisioningService.
 	results := params.ErrorResults{
 		Results: make([]params.ErrorResult, len(args.FilesystemAttachments)),
 	}
-	one := func(arg params.FilesystemAttachment) error {
-		machineTag, filesystemTag, filesystemAttachmentInfo, err := storagecommon.FilesystemAttachmentToState(arg)
-		if err != nil {
-			return errors.Trace(err)
-		}
-		if !canAccess(machineTag, filesystemTag) {
-			return apiservererrors.ErrPerm
-		}
-		// Check that the machine is provisioned before setting attachment info.
-		machineUUID, err := s.machineService.GetMachineUUID(ctx, machine.Name(machineTag.Id()))
-		if errors.Is(err, machineerrors.MachineNotFound) {
-			return errors.NotFoundf("machine %q", machineTag.Id())
-		}
-		if err != nil {
-			return errors.Trace(err)
-		}
-		_, err = s.machineService.GetInstanceID(ctx, machineUUID)
-		if errors.Is(err, machineerrors.NotProvisioned) {
-			return apiservererrors.ServerError(errors.NotProvisionedf("machine %s", machineTag.Id()))
-		}
-		if err != nil {
-			return errors.Trace(err)
-		}
-		err = s.sb.SetFilesystemAttachmentInfo(machineTag, filesystemTag, filesystemAttachmentInfo)
-		if errors.Is(err, errors.NotFound) {
-			return apiservererrors.ErrPerm
-		}
-		return errors.Trace(err)
-	}
-	for i, arg := range args.FilesystemAttachments {
-		err := one(arg)
-		results.Results[i].Error = apiservererrors.ServerError(err)
-	}
 	return results, nil
 }
 
 // AttachmentLife returns the lifecycle state of each specified machine
 // storage attachment.
 func (s *StorageProvisionerAPIv4) AttachmentLife(ctx context.Context, args params.MachineStorageIds) (params.LifeResults, error) {
```

Comment:
Just adding a PR note here that JIRA-8210 which is under way at the moment is adding support for life back in to the facade off of Dqlite.